codebundles/aws-c7n-ec2-health/sli.robot

```diff
@@ -0,0 +1,99 @@
+    ...    secret__aws_secret_access_key=${AWS_SECRET_ACCESS_KEY}
+    ${count}=     RW.CLI.Run Cli
+    ...    cmd=cat ${OUTPUT_DIR}/aws-c7n-ec2-health/unpatched-ec2-instances/metadata.json | jq '.metrics[] | select(.MetricName == "ResourceCount") | .Value'
+    Log    ${count}
+    ${unpatched_ec2_instances_event_score}=    Evaluate    1 if int(${count.stdout}) <= int(${EVENT_THRESHOLD}) else 0
+    Set Global Variable    ${unpatched_ec2_instances_event_score}
+
+Check for unused AWS EC2 instances in AWS Region `${AWS_REGION}` in AWS account `${AWS_ACCOUNT_ID}` 
+    [Documentation]  Check for unused EC2 instances in AWS Region. 
+    [Tags]    ec2    instance    aws    compute
+    ${result}=    CloudCustodian.Core.Generate Policy   
+    ...    ${CURDIR}/unused-ec2-instances.j2    
+    ...    days=${AWS_EC2_AGE}        
+    ...    tags=${AWS_EC2_TAGS}
+    ${c7n_output}=    RW.CLI.Run Cli
+    ...    cmd=custodian run -r ${AWS_REGION} --output-dir ${OUTPUT_DIR}/aws-c7n-ec2-health ${CURDIR}/unused-ec2-instances.yaml --cache-period 0
+    ...    secret__aws_access_key_id=${AWS_ACCESS_KEY_ID}
+    ...    secret__aws_secret_access_key=${AWS_SECRET_ACCESS_KEY}
+    ${count}=     RW.CLI.Run Cli
+    ...    cmd=cat ${OUTPUT_DIR}/aws-c7n-ec2-health/unused-ec2-instances/metadata.json | jq '.metrics[] | select(.MetricName == "ResourceCount") | .Value'
+    Log    ${count}
+    ${unused_ec2_instances_event_score}=    Evaluate    1 if int(${count.stdout}) <= int(${EVENT_THRESHOLD}) else 0
+    Set Global Variable    ${unused_ec2_instances_event_score}
+
+
+Generate Health Score
+    ${ebs_health_score}=      Evaluate  (${unpatched_ec2_instances_event_score} + ${unused_ec2_instances_event_score}) / 2
+    ${health_score}=      Convert to Number    ${ebs_health_score}  2
+    RW.Core.Push Metric    ${health_score}
+
+** Keywords ***
+Suite Initialization
+    ${AWS_REGION}=    RW.Core.Import User Variable    AWS_REGION
+    ...    type=string
+    ...    description=AWS Region
+    ...    pattern=\w*
+    ${AWS_ACCOUNT_ID}=    RW.Core.Import User Variable   AWS_ACCOUNT_ID
+    ...    type=string
+    ...    description=AWS Account ID
+    ...    pattern=\w*
+    ${AWS_ACCESS_KEY_ID}=    RW.Core.Import Secret   AWS_ACCESS_KEY_ID
+    ...    type=string
+    ...    description=AWS Access Key ID
+    ...    pattern=\w*
+    ${AWS_SECRET_ACCESS_KEY}=    RW.Core.Import Secret   AWS_SECRET_ACCESS_KEY
+    ...    type=string
+    ...    description=AWS Access Key Secret
+    ...    pattern=\w*
+    ${EVENT_THRESHOLD}=    RW.Core.Import User Variable    EVENT_THRESHOLD
+    ...    type=string
+    ...    description=The minimum number of EC2 instance to consider.
+    ...    pattern=^\d+$
+    ...    example=2
+    ...    default=0
+    ${AWS_EC2_AGE}=    RW.Core.Import User Variable    AWS_EC2_AGE
+    ...    type=string
+    ...    description=The age of AWS_EC2 instances in days to consider for filtering.
```

Comment:
This description can be improved - this is for a human to read so that they understand what they are configuring. 

"The age (in days) for EC2 instances to be considered stale."